frontend/src/js/components/subscription/SubscriptionPage.tsx

```diff
@@ -0,0 +1,393 @@
+  const [previewPrice, setPreviewPrice] = useState<PreviewPrice>();
+  const [order, setOrder] = useState();
+  const [contactReason, setContactReason] = useState<ContactReason | ''>('');
+  const [inputHelperText, setInputHelperText] = useState<string>(`The minimum limit for ${selectedPlan.name} is ${selectedPlan.minimalDeviceCount}`);
+  const [limit, setLimit] = useState<number>(selectedPlan.minimalDeviceCount);
+  const [enterpriseMessage, setEnterpriseMessage] = useState('');
+  const stripeAPIKey = useSelector(getStripeKey);
+  const [isPreviewLoading, setIsPreviewLoading] = useState(false);
+  const [showUpgradeDrawer, setShowUpgradeDrawer] = useState(false);
+  const [loadingFinished, setLoadingFinished] = useState(!stripeAPIKey);
+
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+  const currentDeviceLimit = useSelector(getDeviceLimit);
+  const hasCurrentPricing = useSelector(getHasCurrentPricing);
+  const org = useSelector(getOrganization);
+  const { addons: orgAddOns = [], plan: currentPlan = PLANS.os.id as AvailablePlans, trial: isTrial = true, id: orgId } = org;
+  const isOrgLoaded = !!orgId;
+  const plan = Object.values(PLANS).find(plan => plan.id === (isTrial ? PLANS.os.id : currentPlan)) || PLANS.os;
+  const enabledAddons = useMemo(() => orgAddOns.filter(addon => addon.enabled), [orgAddOns]);
+  const currentPlanId = plan.id;
+  const debouncedLimit = useDebounce(limit, TIMEOUTS.debounceDefault);
+
+  // redirect customers paying old price to billing page to see the alert
+  useEffect(() => {
+    if (org.id && !isTrial && !hasCurrentPricing) {
+      navigate('/settings/billing');
+    }
+  }, [isTrial, navigate, hasCurrentPricing, org]);
+
+  //Fetch Billing profile & subscription
+  useEffect(() => {
+    dispatch(getCurrentCard());
+    dispatch(getUserBilling());
+    dispatch(getUserSubscription());
+  }, [dispatch]);
+
+  //Loading stripe Component
+  useEffect(() => {
+    // Make sure to call `loadStripe` outside of a component’s render to avoid recreating
+    // the `Stripe` object on every render - but don't initialize twice.
+    if (!stripePromise) {
+      import(/* webpackChunkName: "stripe" */ '@stripe/stripe-js').then(({ loadStripe }) => {
+        if (stripeAPIKey) {
+          stripePromise = loadStripe(stripeAPIKey).finally(() => setLoadingFinished(true));
+        }
+      });
+    } else {
+      const notStripePromise = new Promise(resolve => setTimeout(resolve, TIMEOUTS.debounceDefault));
+      Promise.race([stripePromise, notStripePromise]).then(result => setLoadingFinished(result !== notStripePromise));
+    }
+  }, [stripeAPIKey]);
+
+  useEffect(() => {
+    if (plan) {
+      setSelectedPlan(plan);
+    }
+
+    if (currentDeviceLimit && limit < currentDeviceLimit) {
+      setLimit(currentDeviceLimit);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [plan, currentDeviceLimit]);
```

Comment:
```suggestion
    setLimit(current => (currentDeviceLimit && current < currentDeviceLimit ? currentDeviceLimit : current));
  }, [plan, currentDeviceLimit]);
```
just to get around the need for the eslint escape

Reply:
It will remove the eslint issue, but will make it less readable IMO, I'll at least name it `updatedLimit` to make it a bit more clear